src/main/java/com/indico/mutation/WorkflowSubmission.java

```diff
@@ -76,15 +87,30 @@ public WorkflowSubmission duplicationId(UUID id){
         JSONArray fileMetadata;
         List<FileInput> files = new ArrayList<FileInput>();
         try {
-            fileMetadata = this.upload(this.files);
-            for (Object f : fileMetadata) {
-                JSONObject uploadMeta = (JSONObject) f;
-                JSONObject meta = new JSONObject();
-                meta.put("name", uploadMeta.getString("name"));
-                meta.put("path", uploadMeta.getString("path"));
-                meta.put("upload_type", uploadMeta.getString("upload_type"));
-                FileInput input = FileInput.builder().filename(((JSONObject) f).getString("name")).filemeta(meta).build();
-                files.add(input);
+            if(this.files != null)
+            {
+                fileMetadata = this.upload(this.files);
+                for (Object f : fileMetadata) {
+                    JSONObject uploadMeta = (JSONObject) f;
+                    JSONObject meta = new JSONObject();
+                    meta.put("name", uploadMeta.getString("name"));
+                    meta.put("path", uploadMeta.getString("path"));
+                    meta.put("upload_type", uploadMeta.getString("upload_type"));
+                    FileInput input = FileInput.builder().filename(((JSONObject) f).getString("name")).filemeta(meta).build();
+                    files.add(input);
+            }
+            }
+            if(this.streams != null){
+                fileMetadata = this.uploadBytes(this.streams);
+                for (Object f : fileMetadata) {
+                    JSONObject uploadMeta = (JSONObject) f;
+                    JSONObject meta = new JSONObject();
+                    meta.put("name", uploadMeta.getString("name"));
+                    meta.put("path", uploadMeta.getString("path"));
+                    meta.put("upload_type", uploadMeta.getString("upload_type"));
+                    FileInput input = FileInput.builder().filename(((JSONObject) f).getString("name")).filemeta(meta).build();
```

Comment:
do we allow them to set the filename from an arg?

Reply:
We do -- they send a map of string -> stream. key is used as a filename.